### ISSUE TITLE
_WD_ElementOptionSelect and ShadowRoots

### ISSUE THREAD

**Issue by Danp2** [Danp2/au3WebDriver#358]
https://www.autoitscript.com/forum/topic/205553-webdriver-udf-help-support-iii/?do=findComment&comment=1505361

**Comment by mlipok** [comment]
Yes I know that we must to check it deeply, thanks for setting this here as an important thing.

**Comment by mlipok** [comment]
Doesn't this link clearly tell us that it works?

https://www.autoitscript.com/forum/topic/205553-webdriver-udf-help-support-iii/?do=findComment&comment=1505239


![image](https://user-images.githubusercontent.com/11089482/222743892-44153476-da3e-49a4-addc-5c2ab72b4f36.png)

![image](https://user-images.githubusercontent.com/11089482/222743965-66dde447-10fa-482b-bd5b-4932911a1dd6.png)

**Comment by Danp2** [comment]
I don't think so. Are you confusing `_WD_ElementSelectAction` with `_WD_ElementOptionSelect`?

**Comment by mlipok** [comment]
https://www.autoitscript.com/forum/topic/205553-webdriver-udf-help-support-iii/?do=findComment&comment=1505364

[@Zul](https://www.autoitscript.com/forum/profile/116565-zul) ` You may be correct. _WD_ElementOptionSelect is the correct choice, expect when a shadow root is involved.`

**Comment by mlipok** [comment]
I will start with snipped posted here:
https://www.autoitscript.com/forum/topic/205553-webdriver-udf-help-support-iii/?do=findComment&comment=1505239

and check if `WD_ElementOptionSelect and ShadowRoots` works or not.

**Comment by Danp2** [comment]
Ok, but I'm fairly confident that I already confirmed that it doesn't work when shadowroots are involved.

**Comment by Danp2** [comment]
`_WD_ElementSelectAction` is designed to interact with an element that was previously located via _WD_FindElement. If you look at the code originally posted by @zul, you'll see that it uses `_WD_FindElement` and `_WD_GetShadowRoot` to access and locate the target element prior to calling `_WD_ElementSelectAction`.

`_WD_ElementOptionSelect` is designed to located then click on a Select element's options. It doesn't yet contain any logic to local Select elements inside a shadow root.
